atlas-java-parent/atlas-java-inspect/src/main/java/io/atlasmap/java/inspect/ClassInspectionService.java

```diff
@@ -755,7 +769,7 @@ protected Integer detectArrayDimensions(Class<?> clazz) {
         return tmpClazz;
     }
     
-    protected List<String> detectParameterizedTypes(Field field) {
+    protected List<String> detectParameterizedTypes(Field field, boolean onlyClasses) {
         List<String> pTypes = null;
         
         if(field != null && field.getGenericType() != null) {
```

Comment:
Just a minor comment which might the code a bit more readable (but please don't change it if you disagree):

I would check the non-conditions all first and return null if the method does not apply to the given field. This would avoid the `if (pTypes == null) ptypes = new ArrayList();` within the loop body.

Sth like:

```java
if (field == null || field.getGenericType() == null || !(field.getGenericType() instanceof ParameterizedType)) {
   return null;
}
Type[] types = ((ParameterizedType) field.getGenericType()).getActualTypeArguments();
if (types.length == 0) {
   return null;
}

pTypes = new ArrayList<String>();
for (Type t : types) {
   // ....

}
return pTypes;
```